src/sonic-eventd/tools/events_tool.cpp

```diff
@@ -0,0 +1,360 @@
+#include <thread>
+#include <stdlib.h>
+#include "events.h"
+#include "events_common.h"
+
+/*
+ * Sample i/p file contents for send
+ *
+ * {"src_0:key-0": {"foo": "bar", "hello": "world" }}
+ * {"src_0:key-1": {"foo": "barXX", "hello": "world" }}
+ *
+ * Repeat the above line to increase entries.
+ * Each line is parsed independently, so no "," expected at the end.
+ */
+
+#define ASSERT(res, m, ...) \
+    if (!(res)) {\
+        int _e = errno; \
+        printf("Failed here %s:%d errno:%d zerrno:%d ", __FUNCTION__, __LINE__, _e, zmq_errno()); \
+        printf(m, ##__VA_ARGS__); \
+        printf("\n"); \
+        exit(-1); }
+
+
+typedef enum {
+    OP_INIT=0,
+    OP_SEND=1,
+    OP_RECV=2,
+    OP_SEND_RECV=3     //SEND|RECV
+} op_t;
+
+
+#define PRINT_CHUNK_SZ 2
+
+/*
+ * Usage:
+ */
+
+const char *s_usage = "\
+-s  - To Send\n\
+-r  - To receive\n\
+Note:\n\
+    when both -s & -r are given:\n\
+        it uses main thread to publish and fork a dedicated thread to receive.\n\
+        The rest of the parameters except -w is used for send\n\
+\n\
+-n  - Count of messages to send/receive. When both given, it is used as count to send\n\
+      Default: 1 \n\
+      A value of 0 implies unlimited\n\
+\n\
+-p  - Count of milliseconds to pause between sends or receives. In send-recv mode, it only affects send.\n\
+      Default: 0 implying no pause\n\
+\n\
+      -i  - List of JSON messages to send in a file, with each event/message\n\
+            declared in a single line. When n is more than size of list, the list\n\
+            is rotated upon completion.\n\
+      e.g. '[ \n\
+                { \"sonic-bgp:bgp-state\": { \"ip\": \"10.101.01.10\", \"ts\": \"2022-10-11T01:02:30.45567\", \"state\": \"up\" }}\n\
+                { \"abc-xxx:yyy-zz\": { \"foo\": \"bar\", \"hello\":\"world\", \"ts\": \"2022-10-11T01:02:30.45567\"}}\n\
+                { \"some-mod:some-tag\": {}}\n\
+            ]\n\
+      Default: <some test message>\n\
+\n\
+-o  - O/p file to write received events\n\
+      Default: STDOUT\n";
+
+
+bool term_receive = false;
+
+template <typename Map>
+string
+t_map_to_str(const Map &m)
```

Comment:
template <typename Map>
string
map_to_str(const Map &m)
{
    stringstream _ss;
    string sep;

    _ss << "{";
    for (const auto elem: m) {
        _ss << sep << "{" << elem.first << "," << elem.second.substr(0,10) << "}";
        if (sep.empty()) {
            sep = ", ";
        }
    }
    _ss << "}";
    return _ss.str();
}

Wouldn't it be possible to reuse this helper method in events_common.h

Reply:
One in events_common.h truncates
There, it is part of error reporting, so truncation is *required*
Here it is a tool which has to dump the entire event.